internal/bus/message_pipe.go

```diff
@@ -62,13 +62,15 @@ func NewMessagePipe(size int) *MessagePipe {
 	}
 }
 
-func (p *MessagePipe) Register(size int, plugins []Plugin) error {
+func (p *MessagePipe) Register(ctx context.Context, size int, plugins []Plugin) error {
 	p.pluginsMutex.Lock()
 	defer p.pluginsMutex.Unlock()
 
 	p.plugins = append(p.plugins, plugins...)
 	p.bus = messagebus.New(size)
 
+	p.initPlugins(ctx)
```

Comment:
Are we sure we want to init **all** the plugins in the message pipe or only the ones that were registered?

Reply:
pushed a fix for that